### PR TITLE
Added support for Breakpad .so files to AppCenterDistributeV3 task

### PR DESCRIPTION
**Pull request by vaagnavanesyan**
To reach HockeyApp parity AppCenter task need support Breakpad .so files uploading.
This PR:
- adds support for Breakpad .so files with new `Native Library File Path` field for Android platform
- fixes `createZipStream` method for case when files paths are provided in `symbolsPaths`
  - fixes existing tests due to codebase was changed
- adds new tests to cover new features

### CODE REVIEW

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 1]
```diff
@@ -404,16 +404,16 @@ function expandSymbolsPaths(symbolsType: string, pattern: string, continueOnErro
...
         // multiple dSYM folder paths for Apple application.
-        let dsymPaths = utils.resolvePaths(pattern, continueOnError, packParentFolder);
+        let symPaths = utils.resolvePaths(pattern, continueOnError, packParentFolder);
 
         // Resolved paths can be null if continueIfSymbolsNotFound is true and the file/folder does not exist.
-        if (dsymPaths) {
-            dsymPaths.forEach(dsymFolder => {
-                if (dsymFolder) {
-                    let folderPath = utils.checkAndFixFilePath(dsymFolder, continueOnError);
+        if (symPaths) {
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: Nitpick: This was named `dsymPaths` before since it only referenced dSYMs - we could name it (and according variables) `symbolPaths` now instead - also using `symbolFolder` instead of `symFolder` below, i.e.

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 2]
```diff
@@ -545,27 +539,37 @@ async function run() {
...
-            let version: string;
-            let build: string;
-            if (symbolsType === "AndroidProguard") {
-                const release = await getRelease(effectiveApiServer, effectiveApiVersion, appSlug, releaseId, apiToken, userAgent);
-                version = release.short_version;
-                build = release.version;
-            }
-            const symbolsUploadInfo = await beginSymbolUpload(effectiveApiServer, effectiveApiVersion, appSlug, symbolsType, apiToken, userAgent, version, build);
+        for (const { fieldName, symbolType, forceArchive } of symbolsLookup) {
+            let symbolsPathPattern: string = tl.getInput(fieldName, false);
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: Can these three (symbolsPathPattern, symbolsPaths, symbolsFile) be `const`s?

File: Tasks/AppCenterDistributeV3/Tests/L0SymDistributeBreakpad.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+tmr.setInput('serverEndpoint', 'MyTestEndpoint');
+tmr.setInput('appSlug', 'testuser/testapp');
+tmr.setInput('app', '/test/path/to/my.apk');
+tmr.setInput('symbolsType', 'Android');
+tmr.setInput('releaseNotesSelection', 'releaseNotesInput');
+tmr.setInput('releaseNotesInput', 'my release notes');
+tmr.setInput('nativeLibrariesPath', '/local/**/*.so')
+
+//prepare upload
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/Tests/L0SymDistributeBreakpad.ts]: Code style nitpick: Typically we add a single space after `//`

File: Tasks/AppCenterDistributeV3/Tests/L0SymMultipleDSYMs_flat_1.ts [hunk 4]
```diff
@@ -181,27 +181,12 @@ fs.readdirSync = (folder: string) => {
 
 fs.statSync = (s: string) => {
     let stat = new Stats;
-
-    stat.isFile = () => {
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/Tests/L0SymMultipleDSYMs_flat_1.ts]: These are good additions, but technically could be part of another PR?

Comment (vaagnavanesyan) [reply Tasks/AppCenterDistributeV3/Tests/L0SymMultipleDSYMs_flat_1.ts]: Reverted changes for this file. It will be moved to separated PR

**Comment by vaagnavanesyan** [comment]
@ranterle all suggested changes was applied and tests simplification was moved to separated PR: #12097 🙂

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 5]
```diff
@@ -404,16 +404,16 @@ function expandSymbolsPaths(symbolsType: string, pattern: string, continueOnErro
 
     let symbolsPaths: string[] = [];
 
-    if (symbolsType === "Apple") {
+    if (symbolsType === "Apple" || symbolsType === "Breakpad") {
         // User can specifay a symbols path pattern that selects
-        // multiple dSYM folder paths for Apple application.
-        let dsymPaths = utils.resolvePaths(pattern, continueOnError, packParentFolder);
+        // multiple symbols folder paths.
+        let symbolPaths = utils.resolvePaths(pattern, continueOnError, packParentFolder);
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: Nit, can this be a `const` or are we modifying it further down the line?

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 6]
```diff
@@ -468,23 +468,23 @@ async function run() {
         "Windows": "Windows 8.1",
         "UWP": "Universal Windows Platform (UWP)"
         */
-        const taskSymbolsType: string = tl.getInput('symbolsType', false);
-        const symbolsType = (taskSymbolsType === 'Android' ? 'AndroidProguard' : taskSymbolsType) as SymbolType;
-        let symbolVariableName = null;
-        switch (symbolsType) {
+        const symbolType: TaskSymbolType = tl.getInput('symbolsType', false) as TaskSymbolType;
+        let symbolsLookup: { symbolType: ApiSymbolType, fieldName: string, forceArchive?: boolean }[] = [];
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: Can this be a `const`?

File: Tasks/AppCenterDistributeV3/appcenterdistribute.ts [hunk 7]
```diff
@@ -296,11 +296,11 @@ function getBranchName(ref: string): string {
  * If the input is a single folder, zip it's content. The archive name is the folder's name
  * If the input is a set of folders or files, zip them so they appear on the root of the archive. The archive name is the parent folder's name.
  */
-function prepareSymbols(symbolsPaths: string[]): Q.Promise<string> {
+function prepareSymbols(symbolsPaths: string[], forceArchive: boolean): Q.Promise<string> {
```
Comment (ranterle) [inline Tasks/AppCenterDistributeV3/appcenterdistribute.ts]: I think it might be better to initialize `forceArchive` with a default parameter here.
